src/cstd.h

```diff
@@ -304,6 +304,46 @@ stdnone_of(void *arr, size_t stride, size_t len, int (boolfunc)(const void *))
   return 1;
 }
 
+// Swaps element a with element b
+// complains if either is null for some reason
+#define __STDSWAP(type)                     \
+        void stdswap_##type(type* _val1, type* _val2) \
+        {                                             \
+          if (_val1 == NULL || _val2 == NULL)         \
+            __STD_PANIC("Error: null pointer detected in stdswap_%s", #type); \
+          type tmp = *_val1;                          \
+          *_val1 = *_val2;                            \
+          *_val2 = tmp;                               \
+        }
+
+
+typedef int (*CompareFunction)(const void *, const void *);
+
+const void 
+*is_sorted_until(const void *first, const void *last, CompareFunction compare) 
```

Comment:
1. Make sure to put "std" in front of the function name.
2. Make sure to put the '*' on the same line as 'const void' since it is apart of the return type.
3. Instead of the typedef, lets just put int (*compare)(const void *, const void *) in the function params. Makes it easier to see what exactly it takes.
4. It might be easier to just have this function take a 'stride' so we know what the size of each elem is.